rest-api-spec/src/main/resources/rest-api-spec/test/mtermvectors/20_deprecated.yml

```diff
@@ -0,0 +1,95 @@
+
+---
+"Deprecated parameters should produce warning in Term Vectors query":
+
+   - skip:
+       version: " - 6.99.99"
+       reason:  versionType, _version_type are deprecated starting from 7.0, version_type is used instead
+       features: "warnings"
+
+   - do:
+       indices.create:
+           index: testidx
+           body:
+               mappings:
+                   testtype:
+                       properties:
+                           text:
+                               type :         "text"
+                               term_vector :  "with_positions_offsets"
+
+   - do:
+       index:
+           index: testidx
+           type:  testtype
+           id:    testing_document
+           body:  {"text" : "The quick brown fox is brown."}
+
+   - do:
+       mtermvectors:
+           "term_statistics" : true
+           "body" :
+               "docs":
+                   -
+                       "_index" : "testidx"
+                       "_type" : "testtype"
+                       "_id" : "testing_document"
+                       "version" : 1
+                       "versionType" : "external"
+       warnings:
+           - "Deprecated field [versionType] used, expected [version_type] instead"
+
+   - do:
+       mtermvectors:
+           "term_statistics" : true
+           "body" :
+               "docs":
+                   -
+                       "_index" : "testidx"
+                       "_type" : "testtype"
+                       "_id" : "testing_document"
+                       "version" : 1
+                       "_version_type" : "external"
+       warnings:
+           - "Deprecated field [_version_type] used, expected [version_type] instead"
+
+
+
+---
+"Deprecated camel case parameters should fail in Term Vectors query":
+
+   - skip:
+       version: " - 7.99.99"
```

Comment:
I don't think we've done that before. I understand the intent but I am not sure we should do that. Maybe add a note in the comments about deprecation and removal ? Blocking a version bump with a test is a bit too much IMO.

Reply:
@nik9000 just saw that you asked for this test. Should we really do that ? If we start adding expectations in the tests for future versions it might take a while to just be able to bump the version.

Reply:
@jimczi This is okay, we do do this as an easy way to track code that needs to be removed in the future (see [#27216](https://github.com/elastic/elasticsearch/pull/27216#discussion_r148426724) for another example). Yes, it does mean that bumping the version to 8.0.0 will take a little more time but I think that's outweighed by the benefit of being able to remove code that should be dead. As long as the assertion/test is covered with a clear comment about what should be removed, I think these removals should be relatively straightforward.

Reply:
Fair enough. Thanks for explaining @jasontedor .  If it's contained and easy to remove I agree that it's a good way to ensure that we do remove the code in the next major version. Let's try to not use this for something else than dead code ;) @mayya-sharipova can you add a note in the reason regarding what needs to be done when the version is bumped.